targets/TARGET_ARM_FM/TARGET_FVP_MPS2/us_ticker.c

```diff
@@ -16,43 +16,69 @@
 #include <stddef.h>
 #include "us_ticker_api.h"
 #include "PeripheralNames.h"
-
 #define US_TICKER_TIMER1      CMSDK_DUALTIMER1
 #define US_TICKER_TIMER2      CMSDK_DUALTIMER2
 #define US_TICKER_TIMER_IRQn  DUALTIMER_IRQn
 
+/** mbed OS HAL API defined us_ticker as an increment ticker
+ *  MPS2 platform provided in SSE-200 are decrement tickers
+ *  with interrupt fired counter reaches 0.
+ *
+ *  So 2 Timers are used to construct mbed OS HAL ticker.
+ *  TIMER1 is for counting, and returns inverted binary when read from it
+ *  TIMER2 is for generate interrupts
+ */
 int us_ticker_inited = 0;
 
 void us_ticker_init(void)
 {
     if (us_ticker_inited) {
+        us_ticker_disable_interrupt();
         return;
     }
-    us_ticker_inited = 1;
 
-    US_TICKER_TIMER1->TimerControl = 0x0; // disable timer
-    US_TICKER_TIMER2->TimerControl = 0x00; // disable timer
-    US_TICKER_TIMER1->TimerLoad = 0xFFFFFFFF;
-    US_TICKER_TIMER2->TimerLoad = 0xFFFFFFFF;
+    US_TICKER_TIMER1->TimerControl  = 0x0ul; // disable TIMER1 and reset all control
+    US_TICKER_TIMER2->TimerControl  = 0x0ul; // disable TIMER2 and reset all control
+
+    US_TICKER_TIMER1->TimerLoad     = 0xFFFFFFFFul;
+    US_TICKER_TIMER2->TimerLoad     = 0xFFFFFFFFul;
+
+    US_TICKER_TIMER1->TimerControl |= CMSDK_DUALTIMER1_CTRL_SIZE_Msk; // set TIMER1 to 32 bit counter
+    US_TICKER_TIMER2->TimerControl |= CMSDK_DUALTIMER2_CTRL_SIZE_Msk; // set TIMER2 to 32 bit counter
 
-    US_TICKER_TIMER1->TimerControl = 0x62; // enable interrupt and set to 32 bit counter and set to periodic mode
-    US_TICKER_TIMER2->TimerControl = 0x42; // enable interrupt and set to 32 bit counter
+    US_TICKER_TIMER1->TimerControl |= CMSDK_DUALTIMER1_CTRL_PRESCALE_Msk; // set TIMER1 with 4 stages prescale
+    US_TICKER_TIMER2->TimerControl |= CMSDK_DUALTIMER2_CTRL_PRESCALE_Msk; // set TIMER2 with 4 stages prescale
 
-    US_TICKER_TIMER1->TimerControl |= 0x80; // enable counter
-    US_TICKER_TIMER2->TimerControl |= 0x80; // enable counter
+    US_TICKER_TIMER2->TimerControl |= CMSDK_DUALTIMER2_CTRL_MODE_Msk;     // set TIMER2 periodic mode
+
+    US_TICKER_TIMER2->TimerControl |= CMSDK_DUALTIMER2_CTRL_ONESHOOT_Msk; // set TIMER2 one-shot mode
+
+    US_TICKER_TIMER1->TimerControl |= CMSDK_DUALTIMER1_CTRL_EN_Msk; // enable TIMER1 counter
+    US_TICKER_TIMER2->TimerControl |= CMSDK_DUALTIMER2_CTRL_EN_Msk; // enable TIMER2 counter
```

Comment:
According to the ticker requirements interrupts should be disabled after init, so I suggest to disable US_TICKER_TIMER2 (I assume that it is used to generating interrupts) and disable interrupts instead of enabling them:
`NVIC_DisableRQ(US_TICKER_TIMER_IRQn);`

Reply:
Ticker interrupts are now disabled, but I think there is no need to enable TIMER2 here. It should be enabled while setting the interrupt.

Reply:
The interrupts and counters ticking controlled by different bits.
Since the interrupts control bit is off. and let the TIMER2 ticking going not having any bad impacts to the HAL ticker behavior in the current implementation.
I can set TIMER2 to NOT ticking by default, reduce the ambiguity.